hpms/src/main/java/gov/cms/ab2d/hpms/processing/HPMSExcelReportProcessor.java

```diff
@@ -0,0 +1,105 @@
+package gov.cms.ab2d.hpms.processing;
+
+import gov.cms.ab2d.common.model.Attestation;
+import gov.cms.ab2d.common.model.Contract;
+import gov.cms.ab2d.common.model.Sponsor;
+import gov.cms.ab2d.common.service.SponsorService;
+
+import org.apache.poi.ss.usermodel.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.io.*;
+import java.time.OffsetDateTime;
+import java.util.Iterator;
+import java.util.Optional;
+
+@Service("hpmsExcelReportProcessor")
+public class HPMSExcelReportProcessor implements ExcelReportProcessor {
+
+    @Autowired
+    private SponsorService sponsorService;
+
+    @Override
+    public void processReport(InputStream xlsInputStream, ExcelType excelType) throws IOException {
+        try (Workbook workbook = excelType.getWorkbookType(xlsInputStream)) {
+            Sheet datatypeSheet = workbook.getSheetAt(0);
+            Iterator<Row> iterator = datatypeSheet.iterator();
+
+            while (iterator.hasNext()) {
+
+                Row currentRow = iterator.next();
+                if (currentRow == null) {
+                    continue;
+                }
+                Cell contractNumberCell = currentRow.getCell(4);
+                if (contractNumberCell == null) {
+                    continue;
+                }
+                // Contract number can occasionally be a numeric cell, so we need to account for that
+                if (contractNumberCell.getCellType() != CellType.STRING) {
+                    continue;
+                }
+                String contractNumber = contractNumberCell.getStringCellValue();
+                if (contractNumber == null) {
+                    continue;
+                }
+                if (contractNumber.startsWith("E") || contractNumber.startsWith("S")) {
+                    String sponsorParentName = currentRow.getCell(0).getStringCellValue();
+                    Double sponsorParentHpmsId = currentRow.getCell(1).getNumericCellValue();
+                    String sponsorName = currentRow.getCell(2).getStringCellValue();
+                    Double sponsorHpmsId = currentRow.getCell(3).getNumericCellValue();
+                    String contractName = currentRow.getCell(5).getStringCellValue();
+
+                    Optional<Sponsor> parentSponsorOptional = sponsorService.findByHpmsIdAndParent(sponsorParentHpmsId.intValue(), null);
+                    Sponsor parentId = null;
+                    if (parentSponsorOptional.isPresent()) {
+                        parentId = parentSponsorOptional.get();
+                    }
+                    Optional<Sponsor> sponsorOptional = sponsorService.findByHpmsIdAndParent(sponsorHpmsId.intValue(),
+                            parentId);
+
+                    Sponsor sponsor;
+                    if (sponsorOptional.isPresent() && sponsorOptional.get().getAttestations().isEmpty() || !sponsorOptional.isPresent()) {
```

Comment:
If the sponsor is there, but has empty attestation, we should still use the existing record, not create a new Sponsor.

Reply:
The empty attestation check was to guard against the case where there's an org with say ID 123, and a parent org that has already been inserted has the same ID 123. If a file gets processed again, it's how I determine if it's just a regular org that's already been processed, or a parent org that already exists, since a parent org should never have any attestations, and a regular org should have at least 1.

Reply:
You could probably restructure the code a bit to deal with the parent first (find-or-create); that would help with the "child" sponsor down the line. As of now, the code is creating the parent last.

Reply:
Ok I think that can work